src/snovault/viewconfigs/searchview.py

```diff
@@ -0,0 +1,320 @@
+    'Target',
+]
+
+
+audit_facets = [
+    ('audit.ERROR.category', {'title': 'Audit category: ERROR'}),
+    ('audit.NOT_COMPLIANT.category', {'title': 'Audit category: NOT COMPLIANT'}),
+    ('audit.WARNING.category', {'title': 'Audit category: WARNING'}),
+    ('audit.INTERNAL_ACTION.category', {'title': 'Audit category: DCC ACTION'})
+]
+
+
+class SearchView(BaseView):
+    def __init__(self, context, request, search_type=None, return_generator=False):
+        super(SearchView, self).__init__(context, request)
+        self.search_type = search_type
+        self.return_generator = return_generator
+
+    def construct_clear_filters(self):
+        searchterm_specs = self.request.params.getall('searchTerm')
+        searchterm_only = urlencode([("searchTerm", searchterm) for searchterm in searchterm_specs])
+        if searchterm_only:
+            # Search term in query string; clearing keeps that
+            clear_qs = searchterm_only
+        else:
+            # Possibly type(s) in query string
+            clear_qs = urlencode([("type", typ) for typ in self.doc_types])
+        return self.request.route_path('search', slash='/') + (('?' + clear_qs) if clear_qs else '')
+
+    def construct_filters(self):
+
+        for item_type in self.doc_types:
+            ti = self.types[item_type]
+            qs = urlencode([
+                (k.encode('utf-8'), v.encode('utf-8'))
+                for k, v in self.request.params.items() if not (k == 'type' and self.types['Item' if v == '*' else v] is ti)
+            ])
+            filters = {
+                'field': 'type',
+                'term': ti.name,
+                'remove': '{}?{}'.format(self.request.path, qs)
+            }
+        return filters
+
+    def construct_doc_types(self, doc_types=None):
+        if doc_types is not None:
+            self.doc_types = doc_types
+            return
+        # Process doc_types
+        if self.search_type is None:
+            if '*' in self.doc_types:
+                self.doc_types = ['Item']
+
+        else:
+            self.doc_types = [self.search_type]
+
+        try:
+            self.doc_types = sorted({self.types[name].name for name in self.doc_types})
+        except KeyError:
+            # Check for invalid types
+            bad_types = [t for t in self.doc_types if t not in self.types]
+            msg = "Invalid type: {}".format(', '.join(bad_types))
+            raise HTTPBadRequest(explanation=msg)
+
+        if not self.doc_types:
+            # For form editing embedded searches
+            if self.request.params.get('mode') == 'picker':
+                self.doc_types = ['Item']
+            # For /search/ with no type= use defalts
+            else:
+                self.doc_types = DEFAULT_DOC_TYPES
+
+    def construct_result_views(self):
+
+        # Add special views like Report and Matrix if search is a single type
+        for item_type in self.doc_types:
+            ti = self.types[item_type]
```

Comment:
This is original code but it messy.  'ti' is set in the for loop then used to in adding views as ti.factory.

This line is equivalent to ti = self.types[-1].  if one exists, which it should at the point give how doc_types is set.  Since it's a factory this probably doesn't matter but then there would be no need to loop through the doc types.  Just set 'ti = self.types[-1]'.  I suggest checking the outcome of ti.factory on different doctypes.  Probably has zero impact.